public/app/features/datasource-drawer/DataSourcePickerWithHistory.test.ts

```diff
@@ -0,0 +1,27 @@
+import { updateHistory } from './DataSourcePickerWithHistory';
+
+describe('DataSourcePickerWithHistory', () => {
+  describe('updateHistory', () => {
+    const early = { uid: 'b', lastUse: '2023-02-27T13:39:08.318Z' };
+    const later = { uid: 'a', lastUse: '2023-02-28T13:39:08.318Z' };
+
+    it('should add an item to the history', () => {
+      expect(updateHistory([], early)).toEqual([early]);
+    });
+
+    it('should sort later entries first', () => {
+      expect(updateHistory([early], later)).toEqual([later, early]);
+    });
+
+    it('should update an already existing history item with the new lastUsed date', () => {
+      const laterB = { uid: 'b', lastUse: later.lastUse };
```

Comment:
It was hard for me to understand what this test was checking. maybe a quick reference could make it clear
```suggestion
      const laterB = { uid: early.uid, lastUse: later.lastUse };
```